crates/pixi_build_frontend/src/tool/mod.rs

```diff
@@ -1,24 +1,37 @@
 mod cache;
+mod installer;
 mod spec;
 
-use std::{collections::HashMap, path::PathBuf};
+use std::{collections::HashMap, path::PathBuf, sync::Arc};
 
-pub use cache::{ToolCacheError, ToolContext};
+pub use cache::ToolCacheError;
 pub use spec::{IsolatedToolSpec, SystemToolSpec, ToolSpec};
 
 use crate::InProcessBackend;
 
+pub use installer::ToolContext;
+
 /// A tool that can be invoked.
 #[derive(Debug)]
 pub enum Tool {
-    Isolated(IsolatedTool),
+    Isolated(Arc<IsolatedTool>),
```

Comment:
Why does this have to be an `Arc` all of a sudden? Would it be possible with some clever moves to remove this?